### PR TITLE
Add `:ObsidianSearch` command

### PR DESCRIPTION
**Pull request by epwalsh**
This will work with either `fzf.vim` or `telescope.nvim`. No additional configuration required.

### CODE REVIEW

**Comment by codingluke** [comment]
Hi @epwalsh, great to see you are so commited to it! :)

I just tired it with https://www.lunarvim.org/ which has **telescope.nvim** included. **fzf.vim** is not installed.

Relevant error: `.../lua/obsidian/command.lua:198 Unknown function: fzf#vim#with_preview`

My settings are:

```lua
  use { 
    "epwalsh/obsidian.nvim",
    commit = "fb09be7adca407aed9d64541ec24d86b47b7e68b",
    config = function()
      require("obsidian").setup {
        dir = "~/Projects/obsidian",
        completion = {
          nvim_cmp = true, -- if using nvim-cmp, otherwise set to false
        }
      }
      require("nvim-treesitter.configs").setup {
        ensure_installed = { "markdown", "markdown_inline" },
        highlight = {
          enable = true,
          additional_vim_regex_highlighting = { "markdown" },
        },
      }
    end
  }
```

Full error log

```
2022-10-11T20:25:36 Notify  ERROR Error executing Lua callback: ...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:162: Vim:E117: Unknown function: fzf#vim#with_preview
stack traceback:
	[C]: in function 'nvim_call_function'
	...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:162: in function 'func'
	...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:199: in function <...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:198>
2022-10-11T20:25:47 Notify  ERROR Error executing Lua callback: ...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:162: Vim:E117: Unknown function: fzf#vim#with_preview
stack traceback:
	[C]: in function 'nvim_call_function'
	...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:162: in function 'func'
	...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:199: in function <...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:198>
2022-10-11T20:25:57 Noice  INFO  <leader>                                                                      <esc> close <bs> go up one level 
2022-10-11T20:25:59 Noice  INFO  <leader> » LunarVim                                                          <esc> close <bs> go up one level 
2022-10-11T20:28:43 Notify  ERROR Error executing Lua callback: ...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:162: Vim:E117: Unknown function: fzf#vim#with_preview
stack traceback:
	[C]: in function 'nvim_call_function'
	...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:162: in function 'func'
	...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:199: in function <...pack/packer/start/obsidian.nvim/lua/obsidian/command.lua:198>
```

File: lua/obsidian/command.lua [hunk 1]
```diff
@@ -144,12 +144,50 @@ command.backlinks = function(client, _)
...
 end
 
+---Search notes.
+---
+---@param client obsidian.Client
+---@param data table
+command.search = function(client, data)
+  local base_cmd = vim.tbl_flatten { util.SEARCH_CMD, { "--column", "--line-number", "--no-heading" } }
+
+  if vim.fn.exists "*fzf#vim#grep" then
```
Comment (codingluke) [inline lua/obsidian/command.lua:154]: I guess here we have to also check for `fzf#vim#with_preview`

Comment (epwalsh) [reply lua/obsidian/command.lua]: Done: https://github.com/epwalsh/obsidian.nvim/pull/27/commits/36cc6c88ddc23031937b772d5077091cd563a247

Comment (codingluke) [reply lua/obsidian/command.lua]: Unfortunately the same error prevails! :( It is to late today, I will check it out tomorrow.

Comment (epwalsh) [reply lua/obsidian/command.lua]: Hmm. Weird. Okay well I changed it so that the check for telescope comes first. That should fix it for you.

**Comment by codingluke** [comment]
The Refactor works (at least for me, the 0 check should be done anyways). However, the search now searches inside all buffer content. This is cool too, on a daily basis a file search would be better ;), IMHO.

**Comment by epwalsh** [comment]
> However, the search now searches inside all buffer content. This is cool too, on a daily basis a filename search would be better ;)

I think searching contents is a better default since some people (like myself) use Zettel IDs for filenames that aren't always easy to search by. But I'm happy to accept a PR with a configuration option to change this behavior.

**Pull request merged**